### PR TITLE
Add CLI for MissionBlueAPI using Click

### LINKED ISSUE #12: We should allow user input options for all params of search_Posts API call
We should probably use [click ](https://click.palletsprojects.com/en/stable/)for this

### PR DESCRIPTION
**Pull request by andewmark**
This pull request introduces a basic CLI for the MissionBlue Module using Click.

Changes:
- Added a CLI that currently echos input input params for testing purposes.
- Ensured the CLI is enhancible (Could use comments to explain code more thoroughly though IMO) and is ready for integration with the actual API in future updates

### CODE REVIEW

File: cli.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    required=False,
+    help='Filter posts by the specified account (omit the @ symbol).'
+)
+
+@click.option(
+    '-l',
+    '--lang',
+    type=str,
+    required=False,
+    help=f'Filter posts by language.\n\nLanguage Options:\n\n {', '.join(list(lang_dict.keys()))}\n\n'
```
Comment (Lementknight) [inline cli.py]: For the format string to work you need to have the {} expression encapsulated within the quotation marks. 
For example:
```
name = "Caleb"
f"Hello my name is {name}"
```
This code is an example of using the format string correctly.

Your code is the equivalent of doing this:
```
f"Hello my name is {", "name}"
```
As your code stands now, it doesn't do that. Which is why you get this error when you run the [python job](https://github.com/Team-Mission-Blue/MissionBlueAPI/actions/runs/12402099086/job/34622910736?pr=15):
image


For the join function call, which string do you want to connect the languages with? Once you figure out which string you want to use to join the list of languages, try replacing line 179 with this code block:
```
help=f'Filter posts by language.\n\nLanguage Options:\n\n {<insert_desired_joiner_string_here>.join(list(lang_dict.keys()))}\n\n'
```

**Review by Lementknight — CHANGES_REQUESTED** [review]

**Comment by Lementknight** [comment]
@andewmark, make sure that you add `click` to the `requirements.txt` file since we are now using a new external Python library.

**Review by Lementknight — COMMENTED** [review]
When I tried to run ```python3 cli.py -q "minecraft"``` before my refactor it kept giving me this error 

```Traceback (most recent call last):
  File "~/MissionBlueAPI/cli.py", line 229, in <module>
    params()
    ~~~~~~^^
  File "~/.virtualenv/lib/python3.13/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
      ~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "~/.virtualenv/lib/python3.13/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "~/.virtualenv/lib/python3.13/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
      ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.virtualenv/lib/python3.13/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "~/MissionBlueAPI/cli.py", line 215, in params
    print(lang_dict[lang])
      ~~~~~~~~~^^^^^^ 
``` 
but now I believe that the code works as intended.

**Review by Lementknight — APPROVED** [review]

**Comment by andewmark** [comment]
I see the issue. I wanted lang to not default to English for our scraper. The issue lies in the way that lang is being used when being echoed within the CLI. When nothing gets passed into lang, it defaults to "", but there is no key/value in lang_dict. 

To solve this I will be adding the key/value "":"" in lang_dict so when click.echo(f"Language: {lang_dict[lang]}") is called it can refer as "".

**Review by Lementknight — APPROVED** [review]
LGTM

**Pull request merged**